### PR TITLE
ARROW-3044: [Python] Remove all occurrences of cython's legacy property definition syntax

### PR DESCRIPTION
**Pull request by kszucs**
plus add missing tests

### CODE REVIEW

File: python/pyarrow/tests/test_ipc.py [hunk 1]
```diff
@@ -340,8 +340,13 @@ def test_message_reader(example_messages):
 
     assert len(messages) == 6
     assert messages[0].type == 'schema'
+    assert isinstance(messages[0].metadata, pa.lib.Buffer)
```
Comment (pitrou) [inline python/pyarrow/tests/test_ipc.py]: Can this simply be `pa.Buffer`?

**Review by pitrou — APPROVED** [review]
Thanks for doing this! Just a question below.

File: python/pyarrow/tests/test_ipc.py (hunk 1 above)
Comment (wesm) [reply python/pyarrow/tests/test_ipc.py]: yes, we should use the public API here

**Comment by wesm** [comment]
Doh, there's a merge conflict now. Will merge after resolved + green build

**Comment by kszucs** [comment]
rebased and green

**Pull request closed without merging**